scalalib/overrides-2.12/scala/Array.scala

```diff
@@ -0,0 +1,717 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala
+
+import scala.collection.generic._
+import scala.collection.{ mutable, immutable }
+import mutable.{ ArrayBuilder, ArraySeq }
+import scala.reflect.ClassTag
+import scala.runtime.ScalaRunTime.{ array_apply, array_update }
+import scala.collection.mutable.WrappedArray
+
+/** Contains a fallback builder for arrays when the element type
+ *  does not have a class tag. In that case a generic array is built.
+ */
+class FallbackArrayBuilding {
+
+  /** A builder factory that generates a generic array.
+   *  Called instead of `Array.newBuilder` if the element type of an array
+   *  does not have a class tag. Note that fallbackBuilder factory
+   *  needs an implicit parameter (otherwise it would not be dominated in
+   *  implicit search by `Array.canBuildFrom`). We make sure that
+   *  implicit search is always successful.
+   */
+  implicit def fallbackCanBuildFrom[T](implicit m: DummyImplicit): CanBuildFrom[Array[_], T, ArraySeq[T]] =
+    new CanBuildFrom[Array[_], T, ArraySeq[T]] {
+      def apply(from: Array[_]) = ArraySeq.newBuilder[T]
+      def apply() = ArraySeq.newBuilder[T]
+    }
+}
+
+/** Utility methods for operating on arrays.
+ *  For example:
+ *  {{{
+ *  val a = Array(1, 2)
+ *  val b = Array.ofDim[Int](2)
+ *  val c = Array.concat(a, b)
+ *  }}}
+ *  where the array objects `a`, `b` and `c` have respectively the values
+ *  `Array(1, 2)`, `Array(0, 0)` and `Array(1, 2, 0, 0)`.
+ *
+ *  @author Martin Odersky
+ *  @since  1.0
+ */
+object Array extends FallbackArrayBuilding {
```

Comment:
Looks like this file should not exist.

Reply:
Actually `Array` in 2.12 was missing, it exists in both 2.11 and 2.13. Probably was removed by mistake in the past. 
It also contains optimizations existing in 2.11 and 2.13. 

However, maybe we should store them as diffs as well? Patch for Array takes ~250 lines where the original file has ~720 lines. What do you think? Same about other not included files like `Predef`, `Symbol`, `package.scala`